freud/density.pyx

```diff
@@ -228,17 +228,33 @@ cdef class GaussianDensity(_Compute):
         """:class:`freud.box.Box`: Box used in the calculation."""
         return freud.box.BoxFromCPP(self.thisptr.getBox())
 
-    def compute(self, system):
+    def compute(self, system, values=None):
         R"""Calculates the Gaussian blur for the specified points.
 
         Args:
             system:
                 Any object that is a valid argument to
                 :class:`freud.locality.NeighborQuery.from_system`.
+            values ((:math:`N_{points}`) :class:`numpy.ndarray`):
+                Values associated with the system points used to calculate the
+                convolution. Calculates Gaussian blur if :code:`None`.
+                (Default value = :code:`None`).
         """
         cdef freud.locality.NeighborQuery nq = \
             freud.locality.NeighborQuery.from_system(system)
-        self.thisptr.compute(nq.get_ptr())
+
+        if values is None:
+            values = freud.util._convert_array(
+                np.ones(nq.points.shape[0]), shape=(nq.points.shape[0], ),
+                dtype=np.float32)
+        else:
+            values = freud.util._convert_array(
+                values, shape=(nq.points.shape[0], ), dtype=np.float32)
+
+        cdef float[::1] l_values = values
```

Comment:
This can be marked as `const`.
```suggestion
        cdef const float[::1] l_values = values
```

Reply:
@bdice thanks a lot for your comments. I have added tests and `nullptr` to the latest commit

Reply:
@bdice Thanks for your valuable comments. I have committed your suggestions about Gaussian blur and double-float conversion, added tests and nullptr